base/db/R/check.missing.files.R

```diff
@@ -0,0 +1,49 @@
+#' Function to check if result has empty or missing files
+#' 
+#' @param result A list of dataframes with file paths
+#' @param outname Name of the output file
+#' @param existing.input Existing input records
+#' @param existing.dbfile Existing dbfile records
+#' @return A list of dataframes with file paths, a list of strings with the output file name, a list of existing input records, and a list of existing dbfile records
+#' 
+#' @author Betsy Cowdery, Michael Dietze, Ankur Desai, Tony Gardella, Luke Dramko
+
+check_missing_files <- function(result, outname, existing.input = NULL, existing.dbfile = NULL) {
+  result_sizes <- purrr::map_dfr(
+    result,
+    ~ dplyr::mutate(
+      .,
+      file_size = purrr::map_dbl(file, file.size),
+      missing = is.na(file_size),
+      empty = file_size == 0
+    )
+  )
+  
+  if (any(result_sizes$missing) || any(result_sizes$empty)) {
+    log_format_df <- function(df) {
+      formatted_df <- rbind(colnames(df), format(df))
+      formatted_text <- purrr::reduce(formatted_df, paste, sep = " ")
+      paste(formatted_text, collapse = "\n")
+    }
+    
+    PEcAn.logger::logger.severe(
+      "Requested Processing produced empty files or Nonexistent files:\n",
+      log_format_df(result_sizes[, c(1, 8, 9, 10)]),
+      "\n Table of results printed above.",
+      wrap = FALSE
+    )
+  }
+  
+  # Insert into Database
+  outlist <- unlist(strsplit(outname, "_"))
+  
+  # Wrap in a list for consistant processing later
+  if (exists("existing.input") && is.data.frame(existing.input)) {
+    existing.input <- list(existing.input)
+  }
+  
+  if (exists("existing.dbfile") && is.data.frame(existing.dbfile)) {
+    existing.dbfile <- list(existing.dbfile)
+  }
+  return(list(result_sizes, outlist, existing.input, existing.dbfile))
```

Comment:
Can you check how many of these values are actually used later? On a quick look I think `result_sizes` is only used on lines 22 and 31 above and thus doesn't need to be returned from this function, but please confirm that before removing.

Reply:
`result_sizes` can be safely removed. It is only utilised within `check.missing.files` to verify if _Requested Processing produced empty files or Nonexistent files_. Similarly, `outlist` is also not utilised after its declaration as follow :

```r
## base/db/R/check.missing.files.R

 # Insert into Database
  outlist <- unlist(strsplit(outname, "_"))
```

`outlist` too is not utilised anytime further. So, that too can be safely deleted since it wasn't used after initialisation.